web/renderer/components/pageComponents/ConnectionsPage/NewConnection/About.tsx

```diff
@@ -1,58 +1,191 @@
+    setState,
+    onSubmit,
+    error: connectErr,
+    storedConnections,
+  } = useConfigContext();
   const { activeTabIndex, setActiveTabIndex } = useTabsContext();
+  const [err, setErr] = useState<Error | undefined>(undefined);
+  const [startDoltServer, setStartDoltServer] = useState(false);
 
   const onNext = () => {
     setActiveTabIndex(activeTabIndex + 1);
   };
 
+  useEffect(() => {
+    if (forElectron) {
+      window.ipc.getDoltServerError(async (msg: string) => {
+        setErr(Error(msg));
+      });
+    }
+  }, []);
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    try {
+      const result = await window.ipc.invoke(
+        "start-dolt-server",
+        state.name,
+        state.port,
+        true,
+      );
+      console.log(result); // "Server started successfully"
+      await onSubmit(e); // Now connect to the server
+    } catch (error) {
+      setErr(Error(`Failed to start Dolt server:, ${error}`));
+      console.error("Failed to start Dolt server:", error);
+    }
+  };
   return (
     <form onSubmit={onNext} className={css.form}>
+      {forElectron && (
+        <Checkbox
+          checked={startDoltServer}
+          onChange={() => {
+            setState({
+              useSSL: startDoltServer,
+              port: startDoltServer ? "" : state.port,
+              isLocalDolt: !startDoltServer,
+            });
+            setStartDoltServer(!startDoltServer);
+          }}
+          name="start-dolt-server"
+          label="Start a dolt server"
+          description="Run a Dolt SQL server hosted directly within Workbench. The app supports only one internal server instance, this restriction does not apply to external Dolt server connections."
+          className={css.checkbox}
+        />
+      )}
       <FormInput
         value={state.name}
-        onChangeString={n => setState({ name: n })}
+        onChangeString={n => {
+          setState({ name: n });
+          setErr(undefined);
+        }}
         label="Name"
         labelClassName={css.label}
         placeholder="my-database (required)"
         light
       />
-      <FormSelect
-        outerClassName={css.typeSelect}
-        className={css.typeSelectInner}
-        labelClassName={css.label}
-        label="Type"
-        val={state.type}
-        onChangeValue={t => {
-          if (!t) return;
-          setState({
-            type: t,
-            port: t === DatabaseType.Mysql ? "3306" : "5432",
-            username: t === DatabaseType.Mysql ? "root" : "postgres",
-          });
-        }}
-        options={[
-          { label: "MySQL/Dolt", value: DatabaseType.Mysql },
-          {
-            label: "Postgres/Doltgres",
-            value: DatabaseType.Postgres,
-          },
-        ]}
-        hideSelectedOptions
-        light
-      />
-      <Button type="submit" disabled={!state.name} className={css.button}>
-        Next
-      </Button>
+      {!startDoltServer && (
```

Comment:
Seems like the components used for starting a dolt server vs not starting a dolt server are pretty different? It might read more clearly to make separate components for those two cases